From c860ac80b0e314cb18796479c0bd5931df693928 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Thu, 7 Sep 2017 20:08:42 +0800
Subject: [PATCH 3326/5242] MLK-17491-31 clk: imx: clk-pfdv2: add error
 checking for invalid pfd index

commit  4cbf394ecdc5aaf4ae2c930da6a9fe22e03c5d5a from
https://source.codeaurora.org/external/imx/linux-imx.git

Give a warning when get an invalid pfd index.

Cc: Bai Ping <ping.bai@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-pfdv2.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/clk/imx/clk-pfdv2.c b/drivers/clk/imx/clk-pfdv2.c
index 6c97fe2..8538f38 100644
--- a/drivers/clk/imx/clk-pfdv2.c
+++ b/drivers/clk/imx/clk-pfdv2.c
@@ -158,6 +158,8 @@ struct clk *imx_clk_pfdv2(const char *name, const char *parent_name,
 	struct clk *clk;
 	struct clk_init_data init;
 
+	WARN_ON(idx > 3);
+
 	pfd = kzalloc(sizeof(*pfd), GFP_KERNEL);
 	if (!pfd)
 		return ERR_PTR(-ENOMEM);
-- 
1.7.9.5

